torch_geometric/nn/models/graphmask_explainer.py

```diff
@@ -0,0 +1,838 @@
+        return_type (str, optional): Denotes the type of output from
+            :obj:`model`. Valid inputs are :obj:`"log_prob"` (the model
+            returns the logarithm of probabilities), :obj:`"prob"` (the
+            model returns probabilities), :obj:`"raw"` (the model returns raw
+            scores) and :obj:`"regression"` (the model returns scalars).
+            (default: :obj:`"log_prob"`)
+        penalty_scaling (int, optional): Scaling value of penalty term. Value
+            must lie between 0 and 10. (default: :obj:`5`)
+        lambda_optimizer_lr (float): The learning rate to optimize the
+            Lagrange multiplier. (default: :obj:`1e-2`)
+        init_lambda (float): The Lagrange multiplier. Value must lie between
+            0 and 1. (default: :obj:`0.55`)
+        allowance (float): A float value between 0 and 1 denotes tolerance
+            level. (default: :obj:`0.03`)
+        feat_mask_type (str, optional): Denotes the type of feature mask
+            that will be learned. Valid inputs are :obj:`"feature"` (a single
+            feature-level mask for all nodes), :obj:`"individual_feature"`
+            (individual feature-level masks for each node), and
+            :obj:`"scalar"` (scalar mask for each node).
+            (default: :obj:`"feature"`)
+        layer_type (str): The type of GNN layer being used in the GNN model.
+            (default: :obj:`GCN`)
+        allow_multiple_explanations: Switch to allow explainer to explain
+            node-level predictions for two or more nodes. Must set to
+            :obj:`False` while explaining graph-level predictions and
+            if only one node-level prediction is to be explained.
+            (default: :obj:`False`)
+        log (bool, optional): If set to :obj:`False`, will not log any
+            learning progress. (default: :obj:`True`)
+        **kwargs (optional): Additional hyper-parameters to override default
+            settings in
+            :attr:`~torch_geometric.nn.models.GraphMaskExplainer.coeffs`.
+    """
+
+    coeffs = {
+        'node_feat_size': 1.0,
+        'node_feat_reduction': 'mean',
+        'node_feat_ent': 0.1,
+    }
+
+    def __init__(self, num_layers, model_to_explain=None, epochs: int = 100,
+                 lr: float = 0.01, num_hops: Optional[int] = 3,
+                 return_type: str = 'log_prob', penalty_scaling: int = 5,
+                 lambda_optimizer_lr: int = 1e-2, init_lambda: int = 0.55,
+                 allowance: int = 0.03, feat_mask_type: str = 'scalar',
+                 layer_type: str = 'GCN',
+                 allow_multiple_explanations: bool = False, log: bool = True,
+                 **kwargs):
+        super().__init__(model_to_explain, lr, epochs, num_hops, return_type,
+                         log)
+        assert feat_mask_type in ['feature', 'individual_feature', 'scalar']
+        assert layer_type in ['GCN', 'GAT', 'FastRGCN']
```

Comment:
Is this neccesary? What property of these are you expecting

Reply:
+1. I don't understand this either.

Reply:
To properly set the feature as well as edge masks inside `set_masks()` according to the layer type and to make this `GraphMaskExplainer` implementation for GNNs with `GAT` layer type error free.